prow/plugins/retitle/retitle.go

```diff
@@ -0,0 +1,128 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package retitle implements the retitle plugin
+package retitle
+
+import (
+	"regexp"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+	"k8s.io/test-infra/prow/plugins/trigger"
+)
+
+const (
+	// pluginName defines this plugin's registered name.
+	pluginName = "retitle"
+)
+
+var (
+	retitleRe = regexp.MustCompile(`(?mi)^/retitle\s*(.*)$`)
+)
+
+func init() {
+	plugins.RegisterGenericCommentHandler(pluginName, handleGenericCommentEvent, helpProvider)
+}
+
+func helpProvider(config *plugins.Configuration, enabledRepos []string) (*pluginhelp.PluginHelp, error) {
+	// The Config field is omitted because this plugin is not configurable.
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: "The retitle plugin allows users to re-title pull requests and issues where GitHub permissions don't allow them to.",
+	}
+	pluginHelp.AddCommand(pluginhelp.Command{
+		Usage:       "/retitle <title>",
+		Description: "Edits the pull request or issue title.",
+		Featured:    true,
+		WhoCanUse:   "Collaborators on the repository.",
+		Examples:    []string{"/retitle New Title"},
+	})
+	return pluginHelp, nil
+}
+
+func handleGenericCommentEvent(pc plugins.Agent, e github.GenericCommentEvent) error {
+	var (
+		org  = e.Repo.Owner.Login
+		repo = e.Repo.Name
+	)
+	return handleGenericComment(pc.GitHubClient, func(user string) (bool, error) {
+		return trigger.TrustedUser(pc.GitHubClient, pc.PluginConfig.TriggerFor(org, repo), user, org, repo)
+	}, pc.Logger, e)
+}
+
+type githubClient interface {
+	CreateComment(owner, repo string, number int, comment string) error
+	GetPullRequest(org, repo string, number int) (*github.PullRequest, error)
+	EditPullRequest(org, repo string, number int, pr *github.PullRequest) (*github.PullRequest, error)
+	GetIssue(org, repo string, number int) (*github.Issue, error)
+	EditIssue(org, repo string, number int, issue *github.Issue) (*github.Issue, error)
+}
+
+func handleGenericComment(gc githubClient, isTrusted func(string) (bool, error), log *logrus.Entry, gce github.GenericCommentEvent) error {
+	// Only consider open PRs and issues, and new comments.
+	if gce.IssueState != "open" || gce.Action != github.GenericCommentActionCreated {
+		return nil
+	}
+
+	// Make sure they are requesting a re-title
+	if !retitleRe.MatchString(gce.Body) {
+		return nil
+	}
+
+	var (
+		org    = gce.Repo.Owner.Login
+		repo   = gce.Repo.Name
+		number = gce.Number
+		user   = gce.User.Login
+	)
+
+	trusted, err := isTrusted(user)
+	if err != nil {
+		log.WithError(err).Error("Could not check if user was trusted.")
+		return err
+	}
+	if !trusted {
+		return gc.CreateComment(org, repo, number, plugins.FormatResponseRaw(gce.Body, gce.HTMLURL, user, `Re-titling can only be requested by trusted users.`))
```

Comment:
Can we add some more details on what a trusted user means here? People who aren't familiar with test-infra might not know what it means :grimacing: 

From a k/k perspective, I think something like `...trusted users like repo collaborators` is good enough IMO.